File: reshape_the_matrix.cpp
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& mat, int r, int c) {
        int size=0;
        vector<int> v;
        for(int i=0;i<mat.size();i++){
            vector<int> temp=mat[i];
            size+=temp.size();
            for(int j=0;j<temp.size();j++){
                v.push_back(temp[j]);
            }
        }
        if(r*c != size)
            return mat;
        int index=0;
        vector<vector<int>> ans;
        for(int i=0;i<r;i++){
            vector<int> t;
            for(int j=0;j<c;j++){
                t.push_back(v[index]);
                index++;
            }
            ans.push_back(t);            
        }
        return ans;
    }
};

File: reshape_the_matrix.java
class Solution {
    public int[][] matrixReshape(int[][] mat, int r, int c) {
        int totalElements = mat.length * mat[0].length;
        if (totalElements != r * c || totalElements % r != 0) {
            return mat;
        }
        List<Integer> list=new ArrayList<>();
        for(int i=0;i<mat.length;i++){
            for(int j=0;j<mat[i].length;j++){
                list.add(mat[i][j]);
            }
        }
        int res[][]=new int[r][c];
        
        int temp=0;
        for(int i=0;i<r && list.size()>0;i++){
            for(int j=0;j<c && list.size()>0;j++){
                res[i][j]=list.get(temp++);
            }
        }
        return res;
    }
}